src/components/shared/Terminal.tsx

```diff
@@ -191,10 +322,39 @@ function Terminal(prop: {
   const handleSubmit = (e: React.FormEvent) => {
     e.preventDefault();
     const result = isValid(input);
-    setCommands([...commands, input, ...result.err, ...result.out]);
+    const newHistory = [...inputHistory, input];
+    if (result.err.length > 0 && result.err[0] === 'CLEAR') {
+      setCommands([]);
+    } else {
+      setCommands([...commands, input, ...result.err, ...result.out]);
+    }
+    setInputHistory(newHistory);
+    setInputHistoryIndex(newHistory.length);
     setInput('$ ');
   };
 
+  const handleKeyDown = (e: React.KeyboardEvent<HTMLInputElement>) => {
```

Comment:
For code style, we should make this into a normal function / shouldn't be an arrow (anonymous) function

Reply:
Yeah, will be changing handleChange, handleKeyDown, handleSubmit to match the named functions of the terminal funcs. This should improve code style and make everything more cohesive, good catch.